mlir/include/mlir/Transforms/Passes.td

```diff
@@ -552,4 +552,21 @@ def TopologicalSort : Pass<"topological-sort"> {
   let constructor = "mlir::createTopologicalSortPass()";
 }
 
+def CompositeFixedPointPass : Pass<"composite-fixed-point-pass"> {
+  let summary = "Composite fixed point pass";
+  let description = [{
+    Composite pass runs provided set of passes until fixed point or maximum
+    number of iterations reached.
+  }];
+
+  let options = [
+    Option<"name", "name", "std::string", /*default=*/"\"CompositeFixedPointPass\"",
+      "Composite pass display name">,
+    Option<"pipelineStr", "pipeline", "std::string", /*default=*/"",
+      "Composite pass inner pipeline">,
+    Option<"maxIter", "max-iterations", "unsigned", /*default=*/"10",
```

Comment:
```suggestion
    Option<"maxIter", "max-iterations", "int", /*default=*/"10",
```

Nit: please use `int` everywhere preferably to represent numbers in general.

Reply:
Changed to int and added check for positive values.